src/sidebar/components/menu-item.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const classnames = require('classnames');
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+
+const { onActivate } = require('../util/on-activate');
+
+const SvgIcon = require('./svg-icon');
+
+/**
+ * An item in a dropdown menu.
+ *
+ * Dropdown menu items display an icon, a label and can optionally have a submenu
+ * associated with them.
+ *
+ * When clicked, menu items either open an external link, if the `href` prop
+ * is provided, or perform a custom action via the `onClick` callback.
+ *
+ * The icon can either be an external SVG image, referenced by URL, or a named
+ * icon rendered by an `SvgIcon`.
+ *
+ * For items that have submenus, the `MenuItem` only provides an indicator as
+ * to whether the submenu is open. The container is responsible for displaying
+ * the submenu items beneath the current item.
+ */
+function MenuItem({
+  href,
+  icon,
+  iconAlt,
+  isDisabled,
+  isExpanded,
+  isSelected,
+  isSubmenuVisible,
+  label,
+  onClick,
+  onToggleSubmenu,
+  style,
+}) {
+  const iconClass = 'menu-item__icon';
+  const iconIsUrl = icon && icon.indexOf('/') !== -1;
+  const labelClass = classnames('menu-item__label', {
+    'menu-item__label--submenu': style === 'submenu',
+  });
+
+  return (
+    <div
+      aria-checked={isSelected}
+      className={classnames('menu-item', {
+        'menu-item--submenu': style === 'submenu',
+        'menu-item--shaded': style === 'shaded',
+        'is-disabled': isDisabled,
+        'is-expanded': isExpanded,
+        'is-selected': isSelected,
+      })}
+      role="menuitem"
+      {...onClick && onActivate('menuitem', onClick)}
+    >
+      {icon !== undefined && (
+        <div className="menu-item__icon-container">
+          {icon &&
+            (iconIsUrl ? (
+              <img className={iconClass} alt={iconAlt} src={icon} />
+            ) : (
+              <SvgIcon name={icon} size={10} />
+            ))}
+        </div>
+      )}
+      {href && (
+        <a
+          className={labelClass}
+          href={href}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          {label}
+        </a>
+      )}
+      {!href && <span className={labelClass}>{label}</span>}
+      {typeof isSubmenuVisible === 'boolean' && (
+        <div
+          className="menu-item__toggle"
+          // We need to pass strings here rather than just the boolean attribute
+          // because otherwise the attribute will be omitted entirely when
+          // `isSubmenuVisible` is false.
```

Comment:
You're correct. The attribute needs to be there even when false so that a screen reader will announce that the item is collapsed. If you pass `aria-expanded={false}`, the attribute just gets omitted entirely.